### ISSUE TITLE
Problem running package_coverage on a package

### ISSUE THREAD

**Issue by maelle** [r-lib/covr#322]
I do not get this issue with another package I've just used `package_coverage` on.

Package https://github.com/leonawicz/epubr/

``` r
goodpractice::gp("C:/Users/Maelle/Documents/ropensci/epubr")
#> Preparing: covr
#> Error in seq.default(dots[[1L]][[155L]], dots[[2L]][[155L]]): 'from' must be a finite number
```

Created on 2018-06-08 by the [reprex package](http://reprex.tidyverse.org) (v0.2.0).

<details>

<summary>Session info</summary>

``` r
devtools::session_info()
#> Session info -------------------------------------------------------------
#>  setting  value                       
#>  version  R version 3.5.0 (2018-04-23)
#>  system   x86_64, mingw32             
#>  ui       RTerm                       
#>  language (EN)                        
#>  collate  English_United States.1252  
#>  tz       Europe/Paris                
#>  date     2018-06-08
#> Packages -----------------------------------------------------------------
#>  package      * version    date      
#>  assertthat     0.2.0      2017-04-11
#>  backports      1.1.2      2017-12-13
#>  base         * 3.5.0      2018-04-23
#>  callr          2.0.4      2018-05-15
#>  clisymbols     1.2.0      2017-05-21
#>  compiler       3.5.0      2018-04-23
#>  covr           3.1.0      2018-06-08
#>  crayon         1.3.4      2017-09-16
#>  cyclocomp      1.1.0      2016-09-10
#>  datasets     * 3.5.0      2018-04-23
#>  debugme        1.1.0      2017-10-22
#>  desc           1.2.0      2018-06-06
#>  devtools       1.13.5     2018-02-18
#>  digest         0.6.15     2018-01-28
#>  evaluate       0.10.1     2017-06-24
#>  goodpractice   1.0.2      2018-05-02
#>  graphics     * 3.5.0      2018-04-23
#>  grDevices    * 3.5.0      2018-04-23
#>  htmltools      0.3.6      2017-04-28
#>  httr           1.3.1      2017-08-20
#>  jsonlite       1.5        2017-06-01
#>  knitr          1.20       2018-02-20
#>  lazyeval       0.2.1      2017-10-29
#>  lintr          1.0.2      2017-11-08
#>  magrittr       1.5        2014-11-22
#>  memoise        1.1.0      2017-04-21
#>  methods      * 3.5.0      2018-04-23
#>  praise         1.0.0      2015-08-11
#>  processx       3.1.0      2018-05-15
#>  R6             2.2.2      2017-06-17
#>  rcmdcheck      1.2.1      2016-09-28
#>  Rcpp           0.12.17    2018-05-18
#>  remotes        1.1.1.9000 2018-06-06
#>  rex            1.1.2      2018-01-09
#>  rmarkdown      1.9        2018-03-01
#>  rprojroot      1.3-2      2018-01-03
#>  rstudioapi     0.7        2017-09-07
#>  stats        * 3.5.0      2018-04-23
#>  stringi        1.2.2      2018-05-02
#>  stringr        1.3.1      2018-05-10
#>  tools          3.5.0      2018-04-23
#>  utils        * 3.5.0      2018-04-23
#>  whoami         1.1.2      2017-10-20
#>  withr          2.1.2      2018-03-15
#>  xml2           1.2.0      2018-01-24
#>  xmlparsedata   1.0.1      2018-01-09
#>  yaml           2.1.19     2018-05-01
#>  source                                   
#>  CRAN (R 3.5.0)                           
#>  CRAN (R 3.5.0)                           
#>  local                                    
#>  CRAN (R 3.5.0)                           
#>  CRAN (R 3.5.0)                           
#>  local                                    
#>  Github (r-lib/covr@2637122)              
#>  CRAN (R 3.5.0)                           
#>  CRAN (R 3.5.0)                           
#>  local                                    
#>  CRAN (R 3.5.0)                           
#>  Github (r-lib/desc@4f60833)              
#>  CRAN (R 3.5.0)                           
#>  CRAN (R 3.5.0)                           
#>  CRAN (R 3.5.0)                           
#>  CRAN (R 3.5.0)                           
#>  local                                    
#>  local                                    
#>  CRAN (R 3.5.0)                           
#>  CRAN (R 3.5.0)                           
#>  CRAN (R 3.5.0)                           
#>  CRAN (R 3.5.0)                           
#>  CRAN (R 3.5.0)                           
#>  CRAN (R 3.5.0)                           
#>  CRAN (R 3.5.0)                           
#>  CRAN (R 3.5.0)                           
#>  local                                    
#>  CRAN (R 3.5.0)                           
#>  CRAN (R 3.5.0)                           
#>  CRAN (R 3.5.0)                           
#>  CRAN (R 3.5.0)                           
#>  CRAN (R 3.5.0)                           
#>  Github (r-lib/remotes@e78d84e)           
#>  Github (kevinushey/rex@d7f9c9e)          
#>  CRAN (R 3.4.4)                           
#>  CRAN (R 3.4.3)                           
#>  CRAN (R 3.4.2)                           
#>  local                                    
#>  CRAN (R 3.5.0)                           
#>  CRAN (R 3.5.0)                           
#>  local                                    
#>  local                                    
#>  CRAN (R 3.4.3)                           
#>  CRAN (R 3.4.4)                           
#>  CRAN (R 3.5.0)                           
#>  Github (MangoTheCat/xmlparsedata@3377216)
#>  CRAN (R 3.5.0)
```

</details>

**Comment by gmbecker** [comment]
For the record I see the same behavior (with different exact indices) for switchr (https://github.com/gmbecker/switchr)

I dont' know the internals of covr super well at all, but it the error is throwing from  the print method (covr:::print.coverage) rather than package_coverage, which completes "successfully" (but with an apparently invalid return object). My conjecture is that the problem is related to NAs in the srcrefs for some things:

```
covg <- package_coverage("<path to switchr checkout>")
> covg
Error in seq.default(dots[[1L]][[180L]], dots[[2L]][[180L]]) : 
  'from' must be a finite number
```

```
str(covg)
List of 1916

<<snip>>

 $ lazyRepo.R:158:25:158:31:25:31:158:158                  :List of 3
  ..$ value    : num 0
  ..$ srcref   : 'srcref' int [1:8] 158 25 158 31 25 31 158 158
  .. ..- attr(*, "srcfile")=Classes 'srcfilealias', 'srcfile' <environment: 0x7f87571e17e8> 
  ..$ functions: chr [1:2] "character#PkgManifest" "innerFun"
 $ Utilities.R:NA:NA:NA:NA:NA:NA:NA:NA                     :List of 3
  ..$ value    : num 8
  ..$ srcref   : 'srcref' int [1:8] NA NA NA NA NA NA NA NA
  .. ..- attr(*, "srcfile")=Classes 'srcfilealias', 'srcfile' <environment: 0x7f87571de978> 
  ..$ functions: chr "normalizePath2"
 $ manifest.R:62:5:62:45:5:45:3793:3793                    :List of 3
  ..$ value    : num 4
  ..$ srcref   : 'srcref' int [1:8] 62 5 62 45 5 45 3793 3793
  .. ..- attr(*, "srcfile")=Classes 'srcfilealias', 'srcfile' <environment: 0x7f87571e0f28> 
  ..$ functions: chr "makeManifest"

<<snip>>
 
 [list output truncated]
 - attr(*, "class")= chr "coverage"
 - attr(*, "package")=List of 17
  ..$ package           : chr "switchr"
  ..$ type              : chr "Package"
  ..$ title             : chr "Installing, Managing, and Switching Between Distinct Sets of\nInstalled Packages"
  ..$ version           : chr "0.12.8"
  ..$ author            : chr "Gabriel Becker[aut, cre]"
  ..$ maintainer        : chr "Gabriel Becker <becker.gabriel@gene.com>"
  ..$ copyright         : chr "Genentech Inc"
  ..$ description       : chr "Provides an abstraction for managing, installing,\nand switching between sets of installed R packages. This all"| __truncated__
  ..$ imports           : chr "tools,\nRJSONIO,\nRCurl"
  ..$ depends           : chr "methods"
  ..$ suggests          : chr "BiocInstaller"
  ..$ systemrequirements: chr "git, svn"
  ..$ license           : chr "Artistic-2.0"
  ..$ url               : chr "https://github.com/gmbecker/switchr"
  ..$ bugreports        : chr "https://github.com/gmbecker/switchr/issues"
  ..$ roxygennote       : chr "6.0.1"
  ..$ path              : chr "/Users/beckerg4/gabe/checkedout/switchr"
 - attr(*, "relative")= logi TRUE
> 

```

**Comment by smilesun** [comment]
Same, `Error: 'from' must be a finite number`

**Comment by smilesun** [comment]
```
Error in seq.default(dots[[1L]][[905L]], dots[[2L]][[905L]]) : 
  'from' must be a finite number
```

**Comment by smilesun** [comment]
```
11: stop("'from' must be a finite number")
10: seq.default(dots[[1L]][[905L]], dots[[2L]][[905L]])
9: (function (...) 
   UseMethod("seq"))(dots[[1L]][[905L]], dots[[2L]][[905L]])
8: mapply(FUN = f, ..., SIMPLIFY = FALSE)
7: Map(seq, x$first_line, x$last_line)
6: unlist(Map(seq, x$first_line, x$last_line))
5: unlist(Map(seq, x$first_line, x$last_line)) %||% integer()
4: expand_lines(df)
3: tally_coverage(x, by = by)
2: print.coverage(x)
1: function (x, ...) 
   UseMethod("print")(x)

```

**Comment by mattdowle** [comment]
Same here for data.table.  We haven't had code coverage for 3 months since the covr update from 3.0.1 to 3.1.0 on 18 May.   Reverting to covr 3.0.1 works.  But both CRAN v3.1.0 and `install_github("covr")` as of now, fail with the following : 
```
> require(covr)
Loading required package: covr
> package_coverage()
Error in seq.default(dots[[1L]][[2378L]], dots[[2L]][[2378L]]) : 
  'from' must be a finite number

> debugger()
Message:  Error in seq.default(dots[[1L]][[2378L]], dots[[2L]][[2378L]]) : 
  'from' must be a finite number
Available environments had calls:
1: function (x, ...) 
UseMethod("print")(x)
2: print.coverage(x)
3: tally_coverage(x, by = by)
4: expand_lines(df)
5: unlist(Map(seq, x$first_line, x$last_line)) %||% integer()
6: unlist(Map(seq, x$first_line, x$last_line))
7: Map(seq, x$first_line, x$last_line)
8: mapply(FUN = f, ..., SIMPLIFY = FALSE)
9: (function (...) 
UseMethod("seq"))(dots[[1]][[2378]], dots[[2]][[2378]])
10: seq.default(dots[[1]][[2378]], dots[[2]][[2378]])
11: stop("'from' must be a finite number")

Enter an environment number, or 0 to exit  Selection: 4
Browsing in the environment with call:
   expand_lines(df)
Called from: debugger.look(ind)
Browse[1]> x[is.na(x$first_line),]
           filename functions first_line first_byte last_line last_byte first_column last_column first_parsed last_parsed value
2378 R/data.table.R rbindlist         NA         NA        NA        NA           NA          NA           NA          NA 43914
3074      R/fread.R     fread         NA         NA        NA        NA           NA          NA           NA          NA  2300
3111     R/fwrite.R    fwrite         NA         NA        NA        NA           NA          NA           NA          NA     0
3807     R/setkey.R    setkey         NA         NA        NA        NA           NA          NA           NA          NA    57
Browse[1]> 
```

I found similar report https://github.com/r-lib/covr/issues/317 which looks like it's fixed.  So I made sure I'm using latest dev version of `covr` as of today, but still get the problem.

I should add that we really really love `covr`, completely depend on it, and have never had a problem before. No idea how `covr` works, but it does, and at C level too which is crucial. Thanks!

**Comment by mattdowle** [comment]
Worked around it with this commit : https://github.com/Rdatatable/data.table/commit/90d3736ffc5e172a4e33f4f00bfbc84b898b3b37. It seems very similar to #317 but with a twist that `else if` is involved.  3 of the files in that commit were just one-line minimal changes I whittled it down to.  For `data.table.R` I just peppered it with extra `{}` before `else if` and didn't spend the time to make that minimal.
